File: Assets/Scripts/GameController.js
import UnityEngine.UI;

public class GameController extends MonoBehaviour {
	public var cue:GameObject;
	public var spawnValue:Vector3;
	private var cueCount:int = 2;
	public var startWait : float;
	public var waveWait : float;

	private var score:int;
	public var socreText:Text;

	private var streak:int;
	public var streakText:Text;

	public var goal:int;
	public var goalText:Text;

	private var gameOver:boolean;
	public var gameOverText:Text;
	function Start(){
		InitText();
		SpawnWaves();
	}
	function SpawnWaves(){
		yield WaitForSeconds(startWait);
		while(true){
			for (var i:int = 0; i < cueCount; i++){
			var spawnPosition = new Vector3(Random.Range(-spawnValue.x, spawnValue.x),spawnValue.y,spawnValue.z);
			var spawnRotation:Quaternion = Quaternion.identity;
			Instantiate(cue, spawnPosition, spawnRotation);
			}
			yield WaitForSeconds(waveWait);
		}
	}
	function InitText() {
		score = 0;
		streak = 0;
		UpdateScore();
		UpdateStreak();
		UpdateGoal();
		gameOver = false;
		gameOverText.text = "";
	}
	public function AddScore(value:int) {
		score += value;
		UpdateScore();
	}

	function UpdateScore() {
		socreText.text = "Score:" + score;
	}

	public function AddStreak(){
		streak++;
		UpdateStreak();
		UpdateGoal();
	}
	function UpdateStreak() {
		streakText.text = "STREAK:" + streak;
	}

	function UpdateGoal() {
		goalText.text = "Goal:" + streak + "/" + goal;
	}

	public function GameOver() {
		gameOver = true;
		gameOverText.text = "Game Over!";
	}

}

File: Assets/Scripts/GameController.cs
using UnityEngine.UI;
using UnityEngine;
using System.Collections;
public class GameController:MonoBehaviour {
	public GameObject cue;
	public Vector3 spawnValue;
	private int cueCount;
	public float startWait;
	public float waveWait;

	private int score;
	public Text socreText;

	private int streak;
	public Text streakText;

	public int goal;
	public Text goalText;

	private bool gameOver;
	public Text gameOverText;
	void Start(){
		InitText();
		StartCoroutine(SpawnWaves());
	}
	IEnumerator SpawnWaves(){
		yield return new WaitForSeconds(startWait);
		while(true){
			for (int i = 0; i < cueCount; i++) {
				var spawnPosition = new Vector3(Random.Range(-spawnValue.x, spawnValue.x),spawnValue.y,spawnValue.z);
				Quaternion spawnRotation = Quaternion.identity;
				Instantiate(cue, spawnPosition, spawnRotation);
			}
			yield return new WaitForSeconds(waveWait);
		}
	}
	void InitText() {
		score = 0;
		streak = 0;
		UpdateScore();
		UpdateStreak();
		UpdateGoal();
		gameOver = false;
		gameOverText.text = "";
	}
	public void AddScore(int value) {
		score += value;
		UpdateScore();
	}

	void UpdateScore() {
		socreText.text = "Score:" + score;
	}

	public void AddStreak(){
		streak++;
		UpdateStreak();
		UpdateGoal();
	}
	void UpdateStreak() {
		streakText.text = "STREAK:" + streak;
	}

	void UpdateGoal() {
		goalText.text = "Goal:" + streak + "/" + goal;
	}

	public void GameOver() {
		gameOver = true;
		gameOverText.text = "Game Over!";
	}

}